cpp/memilio/io/README.md

```diff
@@ -140,14 +140,48 @@ To set the value of a parameter from the command line, first type the correspond
+     - To use the Writer to log something the `History` provides the function `void log(const T& t)` to call the `add_record` function of the `Writer` if the logger function `should_log` returns true.
+     - To access the data from the `History` class after logging we provide the function `get_log` to access all records. For this, the lifetime of the `History` has to be as long as one wants to have access to the data, e.g. a history should not be constructed in the function it is called in when data is needed later.
+     - To access data from a specific logger one can use `std::get<x>` where x is the position of the Logger in the template argument list of the `History` object.
+     - Again refer to [this example](../../examples/history.cpp) for a simple implementation of a history object and [this full ABM example](../../simulation/abm.cpp) for a more advanced use case of the History Object with several History objects in use.
+     - As mentioned multiple Loggers can be constructed and this may be necessary if multiple Writer have to be used.
 
-- The `History` class uses `Loggers` to retrieve data from a given input, and a `Writer` to record this data.
+## Working with the History Object
 
-- A `Logger` is a struct with a type `Type` and functions `Type log(const T&)` and `bool should_log(const T&)`. All `Loggers` must be unique types and default constructible/destructible. Their member `should_log` indicates whether to log, while `Type` and `log` determine what is logged. The input for `should_log` and `log` is the same input of type `T` that is given to `History::log`.
+The History Object is providing a way to save data throughout the simulation process. It offers an interface where users can define the data to be saved from a given object using loggers and the method of saving it using writers. Afterward, the user can access this data from the History Object and manipulate it. For a basic logger use case, refer to [this example](../../examples/history.cpp). For an example demonstrating using a logger in the ABM, refer to [this example](../../examples/abm_history_example.cpp).
 
-- The `Writer` defines the type `Data` to store all records (i.e., the return values of `Logger::log`), and the function `template <class Logger> static void add_record(const Logger::Type&, Data&)` to add a new record. `add_record` is used whenever `History::log` was called and `Logger::should_log` is true.
+### Loggers
 
-- The `History` class is templated on the `Writer` that is used to handle the data (e.g., store it into an array), and the `Loggers` that are used to log data. The `Loggers` used for a `History` must be unique.
+The `Logger` struct is a tool for logging data from a given object. Each user-implemented logger must have a `Type` and implement two functions: `Type log(const T&)` and `bool should_log(const T&)`. The input `T` for these functions is the same as the one given to the `History` member-function `History::log`.
```

Comment:
I think it is useful to give an example for T, e.g. a World& when History is used for an ABM Simulation.
Also please add a code block with a Logger sceleton ( i.e. everything the user has to specify replaced by comments, e.g. `using Type = /* type of the record */;`, or use dots or so.